### ISSUE TITLE
Problems with using tianshou to train pettingzoo's waterworld environment.

### ISSUE THREAD

**Issue by boninggogogo** [aai-institute/tianshou#792]
- [ ] I have marked all applicable categories:
    + [x] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [x] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, gym, torch, numpy, sys, pettingzoo
  print(pettingzoo.__version__, tianshou.__version__, gym.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
  ```
Versions:
-->print(pettingzoo.__version__, tianshou.__version__, gym.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
-->1.22.0 0.4.10 0.26.2 1.11.0+cpu 1.22.0 3.8.13 (default, Mar 28 2022, 06:59:08) [MSC v.1916 64 bit (AMD64)] win32

Requesting help, I've been stuck for days!!!
First, I watched "[tianshou's documentation on the pettingzoo environment](https://tianshou.readthedocs.io/en/master/tutorials/tictactoe.html#train-an-marl-agent)", which is what I learned through.
Then, I tried to use tianshou to train [pettingzoo's waterworld environment](https://pettingzoo.farama.org/environments/sisl/waterworld/), but there was a problem that I could not solve.

Next I will show the code I used, please help me to see if I am using tianshou wrongly.
My code：[Click](https://github.com/boninggogogo/tianshou_waterworld/blob/main/trian_waterworld.py)

Only round(step_per_collect*update_per_step)>=1, which means that the error occurs after the step starts collecting.
Once steps are collected everything will lead to the following error:
The error: [Click](https://github.com/boninggogogo/tianshou_waterworld/blob/main/the%20error.txt)

I hope to be able to solve my problem, there is a great possibility that there is a problem with the way I use tianshou, but I really can not find the cause of the problem, I hope to get help, thank you very much!

**Comment by Trinkle23897** [comment]
Could you take a look? @jjshoots @WillDudley

**Comment by Trinkle23897** [comment]
Oh I know the issue. It's the same as previous: https://github.com/thu-ml/tianshou/issues/692#issuecomment-1182744274

The input `batch` has `batch.obs` in `policy.forward` or `policy.learn` function. The network written in `utils/network` assumes `batch.obs` is a tensor or numpy array instead of a structured data like dict. That comes from environment observation. 

The solution is to extract what you want in a tensor format and send it to pytorch network. For example, you can customize your own network to extract the keys from `batch.obs`, or customize `DDPGPolicy` to extract observation and apply mask on action.

**Comment by boninggogogo** [comment]
Thank you, I do so to solve, thank you very much.
